apiserver/facades/controller/caasapplicationprovisioner/provisioner.go

```diff
@@ -120,14 +126,23 @@ func NewStateCAASApplicationProvisionerAPI(ctx facade.Context) (*APIGroup, error
 		PasswordChanger:          common.NewPasswordChanger(st, common.AuthFuncForTagKind(names.ApplicationTagKind)),
 		LifeGetter:               common.NewLifeGetter(st, common.AuthFuncForTagKind(names.ApplicationTagKind)),
 		AgentEntityWatcher:       common.NewAgentEntityWatcher(st, resources, common.AuthFuncForTagKind(names.ApplicationTagKind)),
-		CharmsAPI:                commonCharmsAPI,
-		ApplicationWatcherFacade: common.NewApplicationWatcherFacadeFromState(st, resources, common.ApplicationFilterCAASSidecar),
+		charmInfoAPI:             commonCharmsAPI,
+		appCharmInfoAPI:          appCharmInfoAPI,
+		ApplicationWatcherFacade: common.NewApplicationWatcherFacadeFromState(st, resources, common.ApplicationFilterNone),
```

Comment:
Given both v1 and v2 workers now consume events for all app types now, we can remove the filtering which was only added when v2 charms were introduced.

Reply:
Yeah, I have an item on my todo list and plan to pull this out in a subsequent PR (just to get this one done a bit faster).